activerecord/test/cases/persistence/reload_association_cache_test.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require "cases/helper"
+
+# The test case below is aimed to fix a bug that was noticeable only with a very specific
+# setup of models. The models are fully defined inside the test case so that they are not
+# accidentally changed later causing the bug effects to disappear.
```

Comment:
Let's move this test to where it makes sense. You can define new models by creating new classes in the test/models folder, no need to do what is being done here.

Those models here still leaks to all the other files, so you aren't really isolating those models from other tests

Reply:
Yep, updated as well.